doc/sketch.md

```diff
@@ -0,0 +1,83 @@
+This documents intends to give an overview of the time complexity of the different graph transformations implemented by the various data types existent in Alga.
+
+When specifying the time and memory complexity of graph algorithms, _n_ will denote the number of vertices in the graph, _m_ will denote the number of edges in the graph, and _s_ will denote the _size_ of the corresponding `Graph` expression.
+
+**`removeVertex`**: Removes a vertex from a given graph. In `NonEmpty` graphs, returns `Nothing` if the resulting graph is empty.
+
+| removeVertex       | Time          | Memory |
+|--------------------|---------------|--------|
+| AdjacencyMap       | `O(n * log(n))` | `   ` |
+| Fold               | `O(s)` | `O(s)` |
+| HigherKinded.Class | `O(s)` | `O(s)` |
+| IntAdjacencyMap    | `O(log(n))` | `O(s)` |
+| NonEmpty           | `O(s)` | `   ` |
+| Relation           | `O(n+m)`| `   ` |
```

Comment:
You didn't cover `Graph` in this table -- is this because the implementation simply reuses `HigherKinded.Class.removeVertex`?

I think it would be nice to have an entry for `Graph` as well, since at some point we might change the implementation and it will no longer rely on `HigherKinded.Class`.